crates/wasi-provider/src/states/starting.rs

```diff
@@ -98,10 +98,10 @@ impl Starting {
 #[async_trait::async_trait]
 impl State<PodState> for Starting {
     async fn next(
-        &self,
+        self: Box<Self>,
```

Comment:
Nit: We can probably change the logic in this function so it doesn't need an Arc Mutex anymore (as we are consuming `self`). However, I can do that in a follow up PR